### PR TITLE
INSTUI-4123 Upgrade command-utils to use EcmaScript modules

### PR DESCRIPTION
**Pull request by matyasf**
`eslint-plugin-import` was replaced with `eslint-plugin-import-x` because it does not support the exports field in package.json now, see https://github.com/import-js/eslint-plugin-import/issues/1810 

This necessitated upgrading `typescript-eslint` to v6 

TEST PLAN:
if build/test/chramotic scripts are running this change should be ok

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,14 +87,15 @@
     "danger": "^11.3.1",
     "esbuild": "^0.21.3",
     "eslint": "^8.57.0",
+    "@typescript-eslint/eslint-plugin": "^6.21.0",
     "husky": "^9.0.11",
```
Comment (matyasf) [inline package.json:91]: for this new version this needs to be in this top place for some reason

File: packages/command-utils/package.json [hunk 2]
```diff
@@ -3,7 +3,11 @@
   "version": "9.1.0",
   "description": "Node CLI utilities made by Instructure Inc.",
   "author": "Instructure, Inc. Engineering and Product Design",
-  "main": "./lib/index.js",
+  "type": "module",
+  "exports": {
+    ".": "./lib/index.js",
+    "./package.json": "./package.json"
+  },
```
Comment (matyasf) [inline packages/command-utils/package.json:10]: This is how a standards compliant package using ESM should look like. In the future sadly we will have to add the legacy "module" field https://stackoverflow.com/questions/42708484/what-is-the-module-package-json-field-for

File: package.json (hunk 1 above)
Comment (joyenjoyer) [reply package.json:91]: What do you mean by in this top place?

Comment (matyasf) [reply package.json:91]: I mean in the top package.json

**Review by balzss — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -71,7 +71,7 @@
     "@commitlint/config-conventional": "^17.8.1",
     "@emotion/cache": "^11.11.0",
     "@emotion/react": "^11.11.4",
-    "@instructure/ui-scripts": "8",
+    "@instructure/ui-scripts": "9",
```
Comment (balzss) [inline package.json:74]: is this dependency even needed?

Comment (balzss) [reply package.json:74]: probably, or before we "unified" lots of our scripts. other question: does `command-utils` need to be a public package?

Comment (matyasf) [reply package.json]: I've removed this dependency from here.

> probably, or before we "unified" lots of our scripts. other question: does command-utils need to be a public package?

nope, but I'll need to double check

Comment (matyasf) [reply package.json]: IMO we should set everything thats possible to private in a future pr

**Review by joyenjoyer — APPROVED** [review]
nice job

**Pull request merged**